iOS/Controllers/HomeworkList/HomeworkSubmitViewController.swift

```diff
@@ -0,0 +1,363 @@
+
+        self.present(alertController, animated: true)
+    }
+}
+
+extension HomeworkSubmitViewController: UITableViewDelegate {
+
+}
+
+extension HomeworkSubmitViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.files.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let fileId = self.files[indexPath.row]
+        if let cell = tableView.dequeueReusableCell(withIdentifier: "fileCell") {
+            let color: UIColor
+            let filename: String
+
+            if let file = self.writableSubmission.files.first(where: { $0.id == fileId }) {
+                if self.submission.files.contains(where: { $0.id == fileId }) {
+                    color = UIColor.black
+                } else {
+                    color = UIColor.green
+                }
+
+                filename = file.name
+            } else {
+                color = UIColor.red
+                filename = self.submission.files.first { $0.id == fileId }?.name ?? ""
+            }
+
+            cell.textLabel?.text = filename
+            cell.textLabel?.textColor = color
+            return cell
+        }
+
+        fatalError("No cell found")
+    }
+
+    func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
+
+        let removeAction = UITableViewRowAction(style: .destructive, title: "Remove") {[unowned self] _, indexPath in
+            let fileID = self.files[indexPath.row]
+            if let file = self.writableSubmission.files.first(where: { $0.id == fileID }) {
+                self.unlink(file: file, from: self.submission)
+            }
+
+            self.updateState()
+        }
+
+        return [removeAction]
+    }
+}
+
+// MARK: - Submission writting
+extension HomeworkSubmitViewController {
+    func link(file: File, to submission: Submission) {
+        let fileObjectId = file.objectID
+        self.writingContext.performAndWait {
+            let file = self.writingContext.typedObject(with: fileObjectId) as File
+            self.writableSubmission.files.insert(file)
+        }
+    }
+
+    fileprivate  func unlink(file: File, from submission: Submission) {
+        let fileObjectID = file.objectID
+        self.writingContext.performAndWait {
+            let file = self.writingContext.typedObject(with: fileObjectID) as File
+            self.writableSubmission.files.remove(file)
+        }
+    }
+
+    fileprivate func write(content: String, to submission: Submission) {
+        self.writingContext.performAndWait { [unowned self] in
+            self.writableSubmission.comment = content
+        }
+    }
+}
+
+// MARK: Image Picker
+extension HomeworkSubmitViewController: UINavigationControllerDelegate {
```

Comment:
empty?

Reply:
This is needed to be a `UIImagePickerControllerDelegate` conforming class